#!/usr/bin/env python
PACKAGE = "jn0_arm_common"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# NOTE: Default values from left arm as of 2011/11/22.

gen.add("gc_x_0", double_t, 0,"Center of gravity of the link 0 : x value",  0.0000, -0.5, 0.5)
gen.add("gc_y_0", double_t, 0,"Center of gravity of the link 0 : y value",  0.0312, -0.5, 0.5)
gen.add("gc_z_0", double_t, 0,"Center of gravity of the link 0 : z value",  0.0299, -0.5, 0.5)
gen.add("mass_0", double_t, 0,"Mass of the link 0", 1.5021, 0, 5)

gen.add("gc_x_1", double_t, 0,"Center of gravity of the link 1 : x value", -0.0069, -0.5, 0.5)
gen.add("gc_y_1", double_t, 0,"Center of gravity of the link 1 : y value",  0.0392, -0.5, 0.5)
gen.add("gc_z_1", double_t, 0,"Center of gravity of the link 1 : z value", -0.0083, -0.5, 0.5)
gen.add("mass_1", double_t, 0,"Mass of the link 1", 1.7708, 0, 5)

gen.add("gc_x_2", double_t, 0,"Center of gravity of the link 2 : x value",  0.0138, -0.5, 0.5)
gen.add("gc_y_2", double_t, 0,"Center of gravity of the link 2 : y value", -0.0055, -0.5, 0.5)
gen.add("gc_z_2", double_t, 0,"Center of gravity of the link 2 : z value",  0.1018, -0.5, 0.5)
gen.add("mass_2", double_t, 0,"Mass of the link 2", 3.1883, 0, 5)

gen.add("gc_x_3", double_t, 0,"Center of gravity of the link 3 : x value",  0.0011, -0.5, 0.5)
gen.add("gc_y_3", double_t, 0,"Center of gravity of the link 3 : y value",  0.1267, -0.5, 0.5)
gen.add("gc_z_3", double_t, 0,"Center of gravity of the link 3 : z value",  0.0057, -0.5, 0.5)
gen.add("mass_3", double_t, 0,"Mass of the link 3", 1.3482, 0, 5)

gen.add("gc_x_4", double_t, 0,"Center of gravity of the link 4 : x value",  0.0006, -0.5, 0.5)
gen.add("gc_y_4", double_t, 0,"Center of gravity of the link 4 : y value",  0.0021, -0.5, 0.5)
gen.add("gc_z_4", double_t, 0,"Center of gravity of the link 4 : z value",  0.0007, -0.5, 0.5)
gen.add("mass_4", double_t, 0,"Mass of the link 4", 0.4095, 0, 5)

exit(gen.generate(PACKAGE, "jn0_arm_common", "MassProperties"))
